src/Core/src/Platform/Gtk/MauiDatePicker.cs

```diff
@@ -409,7 +410,7 @@ void ShowPickerWindow()
 
 			if (_pickerWindow is null)
 			{
-				var picker = new DatePickerWindow(Date);
+				var picker = new DatePickerWindow(Date, this.Toplevel as Window);
```

Comment:
@parhamsaremi @webwarrior-ws shouldn't this be a static cast instead of a dynamic (`as`) one?

Reply:
(if not, then you should check for null in the constructor; so either check for null or change to a static cast)

Reply:
I don't think null check is necessary, as `null` is a valid value for `TransientFor` - it will unset current transient window according to [docs](https://docs.gtk.org/gtk3/method.Window.set_transient_for.html).

Reply:
but if you dont assign to null it would have null value anyway so I don't think it's needed?

Reply:
Yes, it will make no difference. So we can leave the code as it is now.

Reply:
I mean if you add a `if (parentWindow != null) { TransientFor = parentWindow; }` you prevent setting null to a property. There's no need to set it when it's null. And this would be more readable.

Reply:
Done.